File: 02_automatedSimplifiedExtensionLayer/00_matlab/functions/get_flow_analysis.m
function [flows_in_processes]=get_flow_analysis(Model)

%% Inputs
for i=2:size(Model.meta_data_flows,1)

flow_analysis_inputs(i,1) = Model.meta_data_flows(i,1);
flow_analysis_inputs(i,2) = Model.meta_data_flows(i,4);
flow_analysis_inputs(i,3) = Model.meta_data_flows(i,2);
flow_analysis_inputs(i,4) = Model.meta_data_flows(i,6);

vector=(Model.matrices.A.mean_values(i-1,1:end)<0)';
col_names=Model.meta_data_processes(1,2:end);
processes= col_names([vector]);

for j=1:length(processes)
    flow_analysis_inputs{i,4+j} = processes{j};
end

end

flow_analysis_inputs{1,1}='flow';
flow_analysis_inputs{1,2}='CAS';
flow_analysis_inputs{1,3}='type';
flow_analysis_inputs{1,4}='unit';
flow_analysis_inputs{1,5}='using processes';
%% Outputs
for i=2:size(Model.meta_data_flows,1)

flow_analysis_outputs(i,1) = Model.meta_data_flows(i,1);
flow_analysis_outputs(i,2) = Model.meta_data_flows(i,14);
flow_analysis_outputs(i,3) = Model.meta_data_flows(i,2);
flow_analysis_outputs(i,4) = Model.meta_data_flows(i,6);

vector=(Model.matrices.A.mean_values(i-1,1:end)>0)';
col_names=Model.meta_data_processes(1,2:end);
processes= col_names([vector]);

for j=1:length(processes)
    flow_analysis_outputs{i,4+j} = processes{j};
end

end

flow_analysis_outputs{1,1}='flow';
flow_analysis_outputs{1,2}='SMILES';
flow_analysis_outputs{1,3}='type';
flow_analysis_outputs{1,4}='unit';
flow_analysis_outputs{1,5}='using processes';

%% save
flows_in_processes.outputs=flow_analysis_outputs;
flows_in_processes.inputs=flow_analysis_inputs;

File: 02_automatedSimplifiedExtensionLayer/utils/get_flow_analysis.py
def get_flow_analysis(Model):
    # Inputs
    flow_analysis_inputs = []  # Initialize list to store input flow analysis

    for i in range(1, len(Model['meta_data_flows'])):  # Start at 1 (Python is 0-based)
        row = [
            Model['meta_data_flows'][i][0],  # flow
            Model['meta_data_flows'][i][3],  # CAS
            Model['meta_data_flows'][i][1],  # type
            Model['meta_data_flows'][i][5]   # unit
        ]

        # Vector for processes with negative values (inputs)
        vector = (Model['matrices']['A']['mean_values'][i-1, :] < 0)
        col_names = Model['meta_data_processes'][0][1:]  # Skip first column
        processes = [col_names[j] for j, v in enumerate(vector) if v]

        # Append processes to the row
        row.extend(processes)
        flow_analysis_inputs.append(row)

    # Add headers for flow_analysis_inputs
    flow_analysis_inputs.insert(0, ['flow', 'CAS', 'type', 'unit', 'using processes'])

    # Outputs
    flow_analysis_outputs = []  # Initialize list to store output flow analysis

    for i in range(1, len(Model['meta_data_flows'])):
        row = [
            Model['meta_data_flows'][i][0],  # flow
            Model['meta_data_flows'][i][13],  # SMILES
            Model['meta_data_flows'][i][1],  # type
            Model['meta_data_flows'][i][5]   # unit
        ]

        # Vector for processes with positive values (outputs)
        vector = (Model['matrices']['A']['mean_values'][i-1, :] > 0)
        col_names = Model['meta_data_processes'][0][1:]  # Skip first column
        processes = [col_names[j] for j, v in enumerate(vector) if v]

        # Append processes to the row
        row.extend(processes)
        flow_analysis_outputs.append(row)

    # Add headers for flow_analysis_outputs
    flow_analysis_outputs.insert(0, ['flow', 'SMILES', 'type', 'unit', 'using processes'])

    # Save inputs and outputs in the result
    flows_in_processes = {
        'inputs': flow_analysis_inputs,
        'outputs': flow_analysis_outputs
    }

    return flows_in_processes
